src/bsoncxx/json.cpp

```diff
@@ -274,15 +274,16 @@ std::string to_json(types::value value) {
 
     json_visitor v(ss, false, 0);
 
-    switch ((int)value.type()) {
-#define BSONCXX_ENUM(name, val)            \
-    case val:                              \
-        v.visit_value(value.get_##name()); \
-        break;
+    if(value.type()) {
```

Comment:
This is a public-facing function, so it's possible somebody could try to use it with an invalid value type:
```
types::value my_value{};
std::cout << "my value: " << bsoncxx::to_json(my_value) << " hooray" << std::endl;
```
I think we should throw an exception if the type isn't set, since that's what we're doing in other public-facing methods like append() above.

Aside, invariant functionality would be handy for other things, we should do it.